include/experimental/__p1673_bits/blas1_dot.hpp

```diff
@@ -59,21 +59,29 @@ struct is_custom_dot_avail : std::false_type {};
 template <class Exec, class v1_t, class v2_t, class Scalar>
 struct is_custom_dot_avail<
   Exec, v1_t, v2_t, Scalar,
-  std::enable_if_t<
-    std::is_same<
-      decltype(dot
-	       (std::declval<Exec>(),
-		std::declval<v1_t>(),
-		std::declval<v2_t>(),
-		std::declval<Scalar>()
-		)
-	       ),
-      Scalar
-      >::value
+  std::void_t<
+    decltype(dot
+	     (std::declval<Exec>(),
+	      std::declval<v1_t>(),
+	      std::declval<v2_t>(),
+	      std::declval<Scalar>()
+	      )
+	     )
     >
   >
 {
-  static constexpr bool value = !std::is_same<Exec,std::experimental::linalg::impl::inline_exec_t>::value;
+  static constexpr bool has_matching_return_type = std::is_same_v<
```

Comment:
Would you consider making `has_matching_return_type` and `is_not_inline_exec_tag` both `private`, as they are implementation details of `value`?